### DISCUSSION TITLE
Unable to build deskflow 1.17.2 in Ubuntu 24.04 and 24.10

### DISCUSSION THREAD

**Discussion by jhonny-oliveira** [deskflow/deskflow#7933] [Development]
Hi!

I'm getting this error building Deskflow 1.17.2 on Ubuntu 24.04 (6.4.2) and 24.10 (QT 6.6.2). The build works fine in the upcoming version of Ubuntu: 25.04 (QT 6.7.2).

```
/<<PKGBUILDDIR>>/src/gui/src/dialogs/AboutDialog.cpp: In constructor ‘AboutDialog::AboutDialog(QWidget*)’:
/<<PKGBUILDDIR>>/src/gui/src/dialogs/AboutDialog.cpp:40:14: error: ‘QIcon::ThemeIcon’ has not been declared
   40 |       QIcon::ThemeIcon::EditCopy, deskflow::gui::isDarkMode() ? QIcon(s_darkCopy) : QIcon(s_lightCopy)
      |              ^~~~~~~~~
```

Any ideas?
You can find the full build log here: 
[deskflow_1.17.2_Ubuntu_24.04_build_log.txt](https://github.com/user-attachments/files/17969731/deskflow_1.17.2_Ubuntu_24.04_build_log.txt).

Thank you!
Jhonny Oliveira
https://xtradeb.net

**Comment by sithlord48** [comment]
You will need Qt 6.7 , to use the themed icons.

**Comment by jhonny-oliveira** [comment]
Is there any chance we could patch it back to support QT 6.4.2 and 6.6.2?

I saw a clause like this somewhere in synergy. It could come in handy, if one knows what to put inside. :-)
```#if QT_VERSION <= QT_VERSION_CHECK(6, 7, 0)```

Thank you!

**Reply by sithlord48** [reply]
We do not want to support Qt < 6.7. After #7934 you will be prevented from configuration if you don't have it.

**Comment by jhonny-oliveira** [comment]
I understand, but that also means that Deskflow won't work on the latest Ubuntu releases: 24.04 (LTS) and 24.10 (interim). :-(

**Reply by sithlord48** [reply]
try the flatpak instead.

**Reply by jhonny-oliveira** [reply]
I leverage only Debian native packages as part of my opensource initiative (https://xtradeb.net). No flatpak or snap. Therefore, I'm a bit stuck here.

I'm not asking the you/the project to maintain retroactive compatibility in the code itself. I'm just asking for a patch, or guidance to build a patch to do so. I will do my best to maintain such patch on my own afterwards.

**Reply by nbolton** [reply]
In case you need Wayland support, your options are:
1. Edit the CMake config to decrease the required lib version for libportal (0.7 doesn't have input capture)
2. [Build and static link libportal](https://github.com/deskflow/deskflow/discussions/7499#discussioncomment-11027553) 0.8 like we did in Deskflow v1.17.0
1. Use/build [v1.17.0](https://github.com/deskflow/deskflow/releases/tag/v1.17.0)

For Qt, maybe v1.17.0 will build ok for you?

**Comment by sithlord48** [comment]
For the Qt / Cmake parts, Here is a small patch for 1.18.0, This should work for 24.10 with Qt 6.6.  You can try with 24.04 but i suspect there is a bit more to change (I've dropped the Qt version to 6.4 so its easier to test there)

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index f992d1614..c092820a4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,7 +94,7 @@ message(STATUS "Building ${CMAKE_PROJECT_NAME}-${CMAKE_PROJECT_VERSION}")
 set(REQUIRED_OPENSSL_VERSION 3.0)
 set(REQUIRED_LIBEI_VERSION 1.3)
 set(REQUIRED_LIBPORTAL_VERSION 0.8)
-set(REQUIRED_QT_VERSION 6.7.0)
+set(REQUIRED_QT_VERSION 6.4.0)
 
 # Control debug item visibility
 # When not set logging is forced to DEBUG and show code locations
diff --git a/src/apps/deskflow-gui/dialogs/AboutDialog.cpp b/src/apps/deskflow-gui/dialogs/AboutDialog.cpp
index f36865aea..1ff2b5d56 100644
--- a/src/apps/deskflow-gui/dialogs/AboutDialog.cpp
+++ b/src/apps/deskflow-gui/dialogs/AboutDialog.cpp
@@ -34,7 +34,7 @@ AboutDialog::AboutDialog(QWidget *parent) : QDialog(parent)
   setWindowTitle(tr("About Deskflow"));
 
   auto copyIcon = QIcon::fromTheme(
-      QIcon::ThemeIcon::EditCopy, deskflow::gui::isDarkMode() ? QIcon(s_darkCopy) : QIcon(s_lightCopy)
+      QStringLiteral("edit-copy"), deskflow::gui::isDarkMode() ? QIcon(s_darkCopy) : QIcon(s_lightCopy)
   );
 
   auto btnCopyVersion = new QPushButton(copyIcon, QString(), this);
diff --git a/src/apps/deskflow-gui/dialogs/ServerConfigDialog.cpp b/src/apps/deskflow-gui/dialogs/ServerConfigDialog.cpp
index 5cc48dfa6..ff6d36b71 100644
--- a/src/apps/deskflow-gui/dialogs/ServerConfigDialog.cpp
+++ b/src/apps/deskflow-gui/dialogs/ServerConfigDialog.cpp
@@ -46,7 +46,7 @@ ServerConfigDialog::ServerConfigDialog(QWidget *parent, ServerConfig &config, Ap
   ui->setupUi(this);
 
   const auto folderIcon =
-      QIcon::fromTheme(QIcon::ThemeIcon::DocumentOpen, QIcon(QStringLiteral(":/icons/64x64/folder.png")));
+      QIcon::fromTheme(QStringLiteral("document-open"), QIcon(QStringLiteral(":/icons/64x64/folder.png")));
   ui->m_pButtonBrowseConfigFile->setIcon(folderIcon);
 
   // force the first tab, since qt creator sets the active tab as the last one
diff --git a/src/apps/deskflow-gui/dialogs/SettingsDialog.cpp b/src/apps/deskflow-gui/dialogs/SettingsDialog.cpp
index 32e8f637a..6eaad3664 100644
--- a/src/apps/deskflow-gui/dialogs/SettingsDialog.cpp
+++ b/src/apps/deskflow-gui/dialogs/SettingsDialog.cpp
@@ -50,7 +50,7 @@ SettingsDialog::SettingsDialog(
   ui->setupUi(this);
 
   const auto folderIcon =
-      QIcon::fromTheme(QIcon::ThemeIcon::DocumentOpen, QIcon(QStringLiteral(":/icons/64x64/folder.png")));
+      QIcon::fromTheme(QStringLiteral("document-open"), QIcon(QStringLiteral(":/icons/64x64/folder.png")));
   ui->m_pPushButtonTlsCertPath->setIcon(folderIcon);
   ui->m_pButtonBrowseLog->setIcon(folderIcon);
```
